pkg/fixtures/test/panic_on_no_return_value_test.go

```diff
@@ -0,0 +1,23 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	mocks "github.com/vektra/mockery/v2/mocks/github.com/vektra/mockery/v2/pkg/fixtures"
+)
+
+func TestPanicOnNoReturnValue(t *testing.T) {
+	m := mocks.NewPanicOnNoReturnValue(t)
+	m.EXPECT().DoSomething()
+
+	defer func() {
+		if r := recover(); r != nil {
+			assert.NotNil(t, r)
+			assert.Equal(t, "Missing Return() function for DoSomething()", r.(string))
+		}
+	}()
+
+	m.DoSomething()
+
```

Comment:
You'll need some sort of assertion that the panic actually happened. Since the recovery is happening within a defer, you might need another deferred statement that asserts the panic occurred. Maybe

Defers are in LIFO order so maybe something like

```go
var panicOccurred bool
defer func() {
    assert.True(t, panicOccurred)
}()
defer func() {
    panicOccurred = true
    // additional recovery logic you already have
}

m.DoSomething()
```